### ISSUE TITLE
Automatically turn double quotes into backticks when starting to write an inline variable

### ISSUE THREAD

**Issue by ferdy-roz** [zed-industries/zed#10661]
### Check for existing issues

- [X] Completed

### Describe the feature

WebStorm has a feature that is honestly a small detail but makes it super nice -- if you start writing an line variable within a double quoted string (e.g. `"${"`) the double quotes get turned into backticks immmediately.

**Comment by Spoutnik97** [comment]
Maybe you should add this precision : it is especially for typescript developers ;)

**Comment by ferdy-roz** [comment]
> Maybe you should add this precision : it is especially for typescript developers ;)

true! done

**Comment by JosephTLyons** [comment]
Yeah - this would be cool for many different languages - PyCharm turns a standard string into an f-string when completing a variable name in between `{}` within the string.

**Comment by autrioly** [comment]
any update on this ?

**Comment by risalfajar** [comment]
Related #33170 #56072
